src/controllers/version.rs

```diff
@@ -5,15 +5,17 @@ pub mod yank;
 
 use super::prelude::*;
 
+use crate::db::DieselPooledConn;
 use crate::models::{Crate, CrateVersions, Version};
 use crate::schema::versions;
 
-fn version_and_crate(req: &mut dyn Request) -> AppResult<(Version, Crate)> {
+fn version_and_crate(req: &dyn Request) -> AppResult<(DieselPooledConn<'_>, Version, Crate)> {
```

Comment:
This function is a little strange now that it returns a connection too, but 🤷‍♀ it's not exactly the cleanest abstraction to begin with anyway given that half the time we don't even use the returned crate. Always more to do.

Reply:
> This function is a little strange now that it returns a connection too

I would be cleaner to obtain a connection and pass it to this function, but it just ~~isn't~~ wasn't possible because of the lifetimes here.  As I write this, I now see that I eventually dropped the `mut`ness of the `&dyn Request`, so it should now be possible to pass in a `&PgConnection` as well, at least I think.

> Always more to do.

Yup!